### PR TITLE
Implement contains function for IP address (continuation)

### PR DESCRIPTION
**Pull request by pellicano**
Developed by: @ebyhr
Contributed by: @pellicano

### CODE REVIEW

**Comment by pellicano** [comment]
@martint signed CLA has been sent through my BlueCat's e-mail.

> Thank you for your pull request and welcome to our community. We require contributors to sign our [Contributor License Agreement](https://github.com/trinodb/cla/raw/master/Trino%20Foundation%20Individual%20CLA.pdf), and we don't seem to have you on file. In order for us to review and merge your code, please submit the signed CLA to [cla@trino.io](mailto:cla@trino.io). For more information, see https://github.com/trinodb/cla.

File: core/trino-main/src/main/java/io/trino/operator/scalar/IpAddressFunctions.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+        return cidrPrefix.equals(addressPrefix);
+    }
+
+    private static boolean isValidIpV6Cidr(int prefixLength)
+    {
+        return prefixLength >= 0 && prefixLength <= 128;
+    }
+
+    private static boolean isValidIpV4Cidr(byte[] address, int prefix)
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/operator/scalar/IpAddressFunctions.java:105]: This method can be simplified to:

```java
long mask = 0xFFFFFFFFL >>> prefix;
return (Ints.fromByteArray(address) & mask) == 0;
```

File: docs/src/main/sphinx/functions/ipaddress.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+====================
+
+
+.. function:: contains(network, address) -> boolean
+    :noindex:
+
+    Returns true if the ``address`` exists in the CIDR ``network``::
+
+        SELECT contains('10.0.0.0/8', IPADDRESS '10.255.255.255'); -- true
+        SELECT contains('10.0.0.0/8', IPADDRESS '11.255.255.255'); -- false
```
Comment (martint) [inline docs/src/main/sphinx/functions/ipaddress.rst:12]: Add an example with IPv6

**Review by martint — CHANGES_REQUESTED** [review]
Can you squash all the commits?

File: core/trino-main/src/main/java/io/trino/operator/scalar/IpAddressFunctions.java (hunk 1 above)
Comment (pellicano) [reply core/trino-main/src/main/java/io/trino/operator/scalar/IpAddressFunctions.java:106]: Done.

**Comment by pellicano** [comment]
@martint commits squashed into 9e6d202217cad9380514a010786921ed4f1ada9d.

**Comment by martint** [comment]
Merged as 4ca945f469532a6d116bcf51f805ac1c6b608a5c. Thanks!

**Pull request closed without merging**